### PR TITLE
Activate DS in MT mode and test its basic functionalites

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx [hunk 1]
```diff
@@ -98,3 +98,39 @@ TEST(TTrivialDS, FromATDFWithJitting)
...
    EXPECT_DOUBLE_EQ(22., *min);
 }
+
+// NOW MT!-------------
+
+TEST(TTrivialDS, FromATDFMT)
+{
+   ROOT::EnableImplicitMT(4);
+   std::unique_ptr<TDataSource> tds(new TTrivialDS(320));
+   TDataFrame tdf(std::move(tds));
```
Comment (amadio) [inline tree/treeplayer/test/dataframe/datasource_trivial.cxx]: I don't understand why not create the tdf as `TDataFrame tdf(TTrivialDS(320));` Any underlying reason?

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx [hunk 2]
```diff
@@ -98,3 +98,39 @@ TEST(TTrivialDS, FromATDFWithJitting)
...
+   auto max2 = tdf.Filter([](ULong64_t col0) { return col0 < 10; }, {"col0"}).Max<ULong64_t>("col0");
+   auto min2 = tdf.Filter([](ULong64_t col0) { return col0 > 10; }, {"col0"})
+                  .Define("j", [](ULong64_t col0) { return col0 * 2; }, {"col0"})
+                  .Min<ULong64_t>("j");
+
+   EXPECT_EQ(320U, *c);
+   EXPECT_DOUBLE_EQ(319., *max);
+   EXPECT_DOUBLE_EQ(0., *min);
+   EXPECT_DOUBLE_EQ(9., *max2);
+   EXPECT_DOUBLE_EQ(22., *min2);
```
Comment (amadio) [inline tree/treeplayer/test/dataframe/datasource_trivial.cxx]: Why `EXPECT_DOUBLE_EQ` on `long`s?

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx (hunk 1 above)
Comment (bluehood) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: `unique_ptr` prevents users from calling methods on a TDS that is being used by TDF, making the interface easier to use right (another minor advantage is that it also lets us support non-copyable TDataSources).

Note that 
```
TDataFrame tdf(TTrivialDS(320));
```
wouldn't work but
```
TDataFrame tdf(new TTrivialDS(320));
```
would, so we're not imposing a lot of extra characters.

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx (hunk 2 above)
Comment (bluehood) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: Unfortunately at the moment `Min` and `Max` always return doubles.

Ideally we would return the exact column type for calls like `Max<T>("x")` and double for jitted calls, `Max("x")`, but I never had the time to implement the required logic.

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx (hunk 1 above)
Comment (dpiparo) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: Side note: we need to have this supported in PyROOT too. The current move is great but we need to "diplomatically" move away from it :-)

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx (hunk 2 above)
Comment (dpiparo) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: I think it's low priority at the moment to iron out this one...

Comment (amadio) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: Here is an example of why I think this is a bad idea:
```
$ root -l
root [0] x = 10101010101010101010.0;
root [1] x
(double) 10101010101010100224.000000
root [2] 
```

Comment (bluehood) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: Yes I agree it is a missing feature

Comment (amadio) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: To put it in other words, ints/longs are exact, converting them to doubles will break that in legitimate situations when numbers are large enough.

Comment (bluehood) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: Icannot but keep agreeing :P PRs are welcome! :)

File: tree/treeplayer/test/dataframe/datasource_trivial.cxx (hunk 1 above)
Comment (bluehood) [reply tree/treeplayer/test/dataframe/datasource_trivial.cxx]: ...I was wrong, users cannot in fact write `TDataFrame tdf(new TTrivialDS(320));` as `unique_ptr`'s ctor overload that takes a raw pointer is marked explicit

File: tutorials/dataframe/tdf010_trivialDataSource.C [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+using TDS = ROOT::Experimental::TDF::TDataSource;
+using TTrivialDS = ROOT::Experimental::TDF::TTrivialDS;
+
+int tdf010_trivialDataSource()
+{
+   auto nEvents = 128U;
+
+   std::unique_ptr<TDS> tds(new TTrivialDS(nEvents));
+   ROOT::Experimental::TDataFrame d_s(std::move(tds));
```
Comment (bluehood) [inline tutorials/dataframe/tdf010_trivialDataSource.C]: should we use the factory function here too?

File: tutorials/dataframe/tdf011_ROOTDataSource.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+'''
+# We prepare an input tree to run on
+fileName = "tdf011_rootDataSource_py.root"
+treeName = "myTree"
+ROOT.gInterpreter.Declare(fill_tree_code)
+ROOT.fill_tree(fileName, treeName)
+
+# Create the data frame
+MakeRootDataFrame = ROOT.ROOT.Experimental.TDF.MakeRootDataFrame
```
Comment (bluehood) [inline tutorials/dataframe/tdf011_ROOTDataSource.py]: missing final `()`?

File: tutorials/dataframe/tdf010_trivialDataSource.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+##
+## \macro_code
+##
+## \date September 2017
+## \author Danilo Piparo
+
+import ROOT
+
+# Create the data frame
+MakeTrivialDataFrame = ROOT.ROOT.Experimental.TDF.MakeTrivialDataFrame
```
Comment (bluehood) [inline tutorials/dataframe/tdf010_trivialDataSource.py]: missing final `()`?

Comment (dpiparo) [reply tutorials/dataframe/tdf010_trivialDataSource.py]: :) no, just a shortcut

File: tutorials/dataframe/tdf010_trivialDataSource.C (hunk 3 above)
Comment (dpiparo) [reply tutorials/dataframe/tdf010_trivialDataSource.C]: not sure. Why not.

File: tutorials/dataframe/tdf010_trivialDataSource.py (hunk 5 above)
Comment (bluehood) [reply tutorials/dataframe/tdf010_trivialDataSource.py]: oh sorry I misread

**Pull request merged**